src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,53 @@
+import { deleteTodos } from '../../api/todos';
+import { useTodos } from '../../context/TodosContext';
+import { TodoFilter } from '../TodoFilter';
+
+export const Footer: React.FC = () => {
+  const { todos, dispatch } = useTodos();
+
+  const uncompletedTodos = todos.filter(todo => !todo.completed).length;
+  const isCompletedTodo = todos.some(todo => todo.completed);
+
+  const { handleDeleteTodo, handleSetError } = useTodos();
+
+  const handleDeletingTodo = async (todoId: number) => {
+    dispatch({ type: 'todos/setIsDeletingAllCompleted', payload: true });
+    try {
+      await deleteTodos(todoId);
```

Comment:
```suggestion
      await deleteTodo(todoId);
```

in this case you delete single todo, not todos